sysbuild/CMakeLists.txt

```diff
@@ -189,6 +189,18 @@ function(${SYSBUILD_CURRENT_MODULE_NAME}_pre_cmake)
       # Make mcuboot a build only target as the main application will flash this from the
       # merged hex file
       set_target_properties(mcuboot PROPERTIES BUILD_ONLY true)
+
+      if(DEFINED board_qualifiers_secure)
+        # Apply configuration for MCUboot using all available RAM or not
+        if(SB_CONFIG_MCUBOOT_USE_ALL_AVAILABLE_RAM)
+          set_config_bool(${DEFAULT_IMAGE} CONFIG_MCUBOOT_USE_ALL_AVAILABLE_RAM y)
+          set_config_bool(mcuboot CONFIG_MCUBOOT_USE_ALL_AVAILABLE_RAM y)
+          set_config_bool(mcuboot CONFIG_MCUBOOT_NRF_CLEANUP_NONSECURE_RAM y)
```

Comment:
nit: This setting is already `default y if MCUBOOT_USE_ALL_AVAILABLE_RAM`. Is this statement needed?

Reply:
I force them both to be set in case one is being checked in some external code. In MCUboot itself, one defaults the other but doesn't force it which is wrong because it should clean it up too